x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/connector_form.tsx

```diff
@@ -108,24 +109,44 @@ const formSerializer = (formData: ConnectorFormSchema): ConnectorFormSchema => {
 
   const webhookFormData = formData as {
     config: { headers?: Array<{ key: string; value: string }> };
+    __internal__?: {
+      headers?: Array<{
+        key: string;
+        value: string;
+        type: 'config' | 'secret';
+      }>;
+    };
   };
-  const headers = (webhookFormData?.config?.headers ?? []).reduce(
-    (acc, header) => ({
-      ...acc,
-      [header.key]: header.value,
-    }),
-    {}
-  );
+
+  const headers = webhookFormData?.__internal__?.headers ?? [];
+
+  const configHeaders = headers
```

Comment:
It seems that the `configHeaders` and `secretHeaders` are constructed with the same logic. Better to move the logic to a util function (even on the same file) that will transform the headers. It will also make the code more readable.